server/bin/pbench-restore-and-unpack-tarballs.py

```diff
@@ -0,0 +1,244 @@
+
+import pbench
+from pbench import (
+    PbenchConfig,
+    BadConfig,
+    get_pbench_logger,
+)
+
+
+_NAME_ = "pbench-restore-and-unpack-tarballs"
+
+tb_pat_r = r"\S+_(\d\d\d\d)[._-](\d\d)[._-](\d\d)[T_](\d\d)[._:](\d\d)[._:](\d\d)\.tar\.xz\.md5"
+tb_pat = re.compile(tb_pat_r)
+
+
+def gen_list(backup):
+    """gen_list - traverse the given BACKUP hierarchy looking for all tar balls
+    that have a .md5 file.
+    """
+    with os.scandir(backup) as backup_scan:
+        for c_entry in backup_scan:
+            if c_entry.name.startswith(".") and c_entry.is_dir(follow_symlinks=False):
+                continue
+            if not c_entry.is_dir(follow_symlinks=False):
+                continue
+            # We have a controller directory.
+            with os.scandir(c_entry.path) as controller_scan:
+                for entry in controller_scan:
+                    if entry.name.startswith(".") and entry.is_dir(
+                        follow_symlinks=False
+                    ):
+                        continue
+                    if entry.is_dir(follow_symlinks=False):
+                        continue
+                    match = tb_pat.fullmatch(entry.name)
+                    if not match:
+                        continue
+                    # Turn the pattern components of the match into a datetime
+                    # object.
+                    tb_dt = datetime(
+                        int(match.group(1)),
+                        int(match.group(2)),
+                        int(match.group(3)),
+                        int(match.group(4)),
+                        int(match.group(5)),
+                        int(match.group(6)),
+                    )
+                    tb = Path(entry.path[:-4])
+                    if not tb.exists():
+                        print(f"what? {entry.path}")
+                        sys.exit(1)
+                    yield tb_dt, c_entry.name, tb
+
+
+def main(options):
+    if not options.cfg_name:
+        print(
+            f"{_NAME_}: ERROR: No config file specified; set"
+            " _PBENCH_SERVER_CONFIG env variable",
+            file=sys.stderr,
+        )
+        return 1
+
+    try:
+        config = PbenchConfig(options.cfg_name)
+    except BadConfig as e:
+        print(f"{_NAME_}: {e}", file=sys.stderr)
+        return 2
+
+    logger = get_pbench_logger(_NAME_, config)
+
+    archive = config.ARCHIVE
+    archive_p = Path(archive).resolve(strict=True)
+
+    if not archive_p.is_dir():
+        logger.error(
+            "The configured ARCHIVE directory, {}, is not a valid directory", archive
+        )
+        return 4
+
+    backup = config.conf.get("pbench-server", "pbench-backup-dir")
+    backup_p = Path(backup).resolve(strict=True)
+
+    if not backup_p.is_dir():
+        logger.error(
+            "The configured pbench-backup-dir directory, {}, is not a valid directory",
+            backup,
+        )
+        return 6
+
+    start = pbench._time()
```

Comment:
Maybe, config.TS?

Reply:
> you cannot subtract the beginning and end TS
Because the value doesn't change once it's initialized?...oh, well -- there are other ways of doing timing...if we really need to.

Reply:
So this is the old way of fixing "time" in unit tests.  You can see this in `pbench-cull-unpacked-tarballs` and `pbench-reindex`.

If we want to fix this, let's fix this on `main`, but I don't think we should be fixing this on `b0.69`, or in this PR.

Reply:
Since this appears to be a brand new module, I'm not sure that arguments about this being an existing way or about this being on an old branch hold much water....  😏